src/Core/Logger/Factory/LegacyLoggerFactory.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+// Copyright (C) 2010-2024, the Friendica project
+// SPDX-FileCopyrightText: 2010-2024 the Friendica project
+//
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+declare(strict_types=1);
+
+namespace Friendica\Core\Logger\Factory;
+
+use Friendica\Core\Config\Capability\IManageConfigValues;
+use Friendica\Core\Hooks\Capability\ICanCreateInstances;
+use Friendica\Util\Profiler;
+use Psr\Log\LoggerInterface;
+
+/**
+ * Manager for the core logging instances
+ */
+final class LegacyLoggerFactory implements LoggerFactory
+{
+	private ICanCreateInstances $instanceCreator;
+
+	private IManageConfigValues $config;
+
+	private Profiler $profiler;
+
+	public function __construct(ICanCreateInstances $instanceCreator, IManageConfigValues $config, Profiler $profiler)
+	{
+		$this->instanceCreator = $instanceCreator;
+		$this->config          = $config;
+		$this->profiler        = $profiler;
+	}
+
+	/**
+	 * Creates and returns a PSR-3 Logger instance.
+	 *
+	 * Calling this method multiple times with the same parameters SHOULD return the same object.
+	 *
+	 * @param \Psr\Log\LogLevel::* $logLevel The log level
+	 * @param \Friendica\Core\Logger\Capability\LogChannel::* $logChannel The log channel
+	 */
+	public function createLogger(string $logLevel, string $logChannel): LoggerInterface
```

Comment:
`$logLevel` isn't used in this method, is this intended?

Reply:
Yes, it is intended. As I stated in point 8.) in the PR description every `LoggerInterface` implementation needs a LogLevel.

 At the moment we have 3 official Logger factories:

- `Friendica\Core\Logger\Factory\StreamLogger`
- `Friendica\Core\Logger\Factory\SyslogLogger`
- and the monolog addon `Friendica\Addon\monolog\src\Factory\Monolog`

All 3 factories are getting the LogLevel by requiring the $config as `Friendica\Core\Config\Capability\IManageConfigValues` implementation to get the logLevel by calling `$config->get('system', 'loglevel')`. By providing the logLevel directly in the factory method all factories could potential get rid of this dependency and a future LoggerFactories might not need this dependency.